src/main/java/org/reactome/server/util/TocPathwayCache.java

```diff
@@ -0,0 +1,42 @@
+package org.reactome.server.util;
+
+import org.reactome.server.graph.domain.result.PathwayResult;
+import org.reactome.server.graph.service.TocService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.Collection;
+
+@Component
+public class TocPathwayCache {
+
+    private static final Logger infoLogger = LoggerFactory.getLogger("infoLogger");
+
+    private TocService tocService;
+
+    private Collection<PathwayResult> tocPathways;
+
+    @Autowired
+    public void setTocService(TocService tocService) {
+        this.tocService = tocService;
+    }
+
+    @PostConstruct
+    public void getTocPathwayResult() {
+
+        try {
+            if (tocPathways == null) {
+                tocPathways = tocService.getAllTocPathway();
+            }
+        } catch (NullPointerException e) {
```

Comment:
same
It is not recommended to catch NPE
https://stackoverflow.com/questions/15146339/catching-nullpointerexception-in-java
why are you catching it ?